src/components/Carousel/Carousel.jsx

```diff
@@ -0,0 +1,239 @@
+      // update the aria-hidden attribute appropriately
+      this.setState(prevState => ({
+        currentSlideIndex:
+          prevState.currentSlideIndex === 0
+            ? prevState.currentSlideIndex
+            : prevState.currentSlideIndex - 1,
+      }));
+      return;
+    }
+    this.setState(prevState => ({
+      left: prevState.left + parseInt(width, 10),
+      currentSlideIndex: prevState.currentSlideIndex - 1,
+    }));
+  };
+
+  moveForward = () => {
+    const { children, width } = this.props;
+    if (this.verifyLastItem(children.length, width)) {
+      // if it's the last item we still want to update the currentSlideIndex so we can
+      // update the aria-hidden attribute appropriately
+      this.setState(prevState => ({
+        currentSlideIndex:
+          prevState.currentSlideIndex === children.length - 1
+            ? prevState.currentSlideIndex
+            : prevState.currentSlideIndex + 1,
+      }));
+      return;
+    }
+
+    this.setState(prevState => ({
+      left: prevState.left - parseInt(width, 10),
+      currentSlideIndex: prevState.currentSlideIndex + 1,
+    }));
+  };
+
+  verifyLastItem = (lengthOfChildren, widthOfEachItem) => {
+    const { left } = this.state;
+    const finalLength = (lengthOfChildren - 1) * parseInt(widthOfEachItem, 10);
+
+    return left === -finalLength;
+  };
+
+  renderItems = () => {
+    const { currentSlideIndex } = this.state;
+    const { children } = this.props;
+    const items = children.map((child, index) => (
+      <MainListItem key={index} aria-hidden={currentSlideIndex !== index}>
+        {child}
+      </MainListItem>
+    ));
+
+    return items;
+  };
+
+  renderAnnouncement = () => {
+    const { currentSlideIndex } = this.state;
+    const { children } = this.props;
+    const content = (
+      <Announcement aria-live="polite" aria-atomic="true" tabIndex={-1}>
+        Slide
+        {currentSlideIndex + 1}
+        out of
+        {children.length}
+      </Announcement>
+    );
+
+    return content;
+  };
+
+  goToSlide = index => {
+    const { width } = this.props;
+    const newLeft = parseInt(width, 10) * index;
+    this.setState({
+      left: -newLeft,
+      currentSlideIndex: index,
+    });
+  };
+
+  render() {
+    const { left, currentSlideIndex } = this.state;
+    const { children, width } = this.props;
+    return (
+      <CarouselStyled>
+        {this.renderAnnouncement()}
+        <ButtonOverlapContainer left>
+          <Button
+            type="secondary"
+            icon={<ArrowLeft />}
+            hasIconOnly
+            onClick={() => {
+              this.moveBackward();
+            }}
+            label="Backwards"
+          />
+        </ButtonOverlapContainer>
+        <Content>
+          <Window width={width}>
+            <MainList left={left}>{this.renderItems()}</MainList>
+          </Window>
+          <IndicatorList>
+            {children.map((child, index) => (
```

Comment:
I think you could use `React.children.map(...)` here instead of children.map (as mentioned in other comments)